app/src/main/java/com/example/fittracker/DbQuery/DbFetchUserData.java

```diff
@@ -0,0 +1,34 @@
+package com.example.fittracker.DbQuery;
+
+import android.os.AsyncTask;
+
+import com.example.fittracker.ConstantUtils;
+import com.example.fittracker.User;
+import com.example.fittracker.UserDao;
+
+import java.util.concurrent.ExecutionException;
+
+public class DbFetchUserData extends AsyncTask<Integer, Void, User> implements DbGenericQuery<User,Integer> {
+    private UserDao userDao;
+
+    public DbFetchUserData(UserDao userDao) {
+        this.userDao = userDao;
+    }
+
+    @Override
+    protected User doInBackground(Integer... ids) {
+        return (userDao.fetchUserData(ids[ConstantUtils.ZERO]));
+    }
+
+    @Override
+    public User executeQuery(Integer id) {
+        try {
+            return this.execute(id).get();
+        } catch (ExecutionException e) {
```

Comment:
You could catch multiple exceptions in a line with the "|" operator.

catch (ExecutionException | InterruptedException e)

Reply:
Done